File: Easy/1518._Water_Bottles/solution.go
// Beats 100% Time 63.64% Space
func numWaterBottles(numBottles int, numExchange int) int {
    count := 0
    for numBottles >= numExchange {
        sets_drank := numBottles/numExchange
        count=count+sets_drank*numExchange
        numBottles = numBottles-(sets_drank*numExchange)+sets_drank
    }
    return count + numBottles
}

File: Easy/1518._Water_Bottles/solution.rs
// Beats 100% Time 100% Space
impl Solution {
    pub fn num_water_bottles(mut num_bottles: i32, num_exchange: i32) -> i32 {
        let mut count:i32 = 0;
        while num_bottles >= num_exchange {
            let mut sets_drank:i32 = num_bottles / num_exchange ;
            count += sets_drank*num_exchange;
            num_bottles = num_bottles-(sets_drank*num_exchange)+sets_drank;
        }
        return count + num_bottles
    }
}
